File: php/PersonagemDirector.php
<?php

use PersonagemBuilder;

class PersonagemDirector{

    private $personagemBuilder;

    public function __construct(PersonagemBuilder $personagemBuilder)
    {
        $this->personagemBuilder = $personagemBuilder;
    }

    public function getPersonagem()
    {
        return $this->personagemBuilder->getPersonagem();
    }

    public function makePersonagem()
    {
            $this->personagemBuilder->buildClasse();
            $this->personagemBuilder->buildEspecie();
            $this->personagemBuilder->buildNome();
    }
}

File: java/PersonagemDirector.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package personagem;

/**
 *
 * @author User
 */
public class PersonagemDirector {
    
    private PersonagemBuilder personagemBuilder;

    public PersonagemDirector(PersonagemBuilder personagemBuilder) {
        this.personagemBuilder = personagemBuilder;
        
    }
   
    public Personagem getPersonagem(){
        return this.personagemBuilder.getPersonagem();
    }
    
    public void makePersonagem(){
        this.personagemBuilder.buildClasse();
        this.personagemBuilder.buildEspecie();
        this.personagemBuilder.buildNome();
    }
    
    
}
